trlx/trainer/accelerate_ppo_trainer.py

```diff
@@ -446,7 +445,7 @@ def make_experience(self, num_rollouts: int = 1024, iter_count: int = 0):  # noq
             # Get the logprobs and values, for tokens that are not padding,
             # from the start of the prompt up to the <eos> token, while also including the latter
             # (these are taken from the student model and not the reference model)
-            ends = start + attention_mask[:, start:].sum(1) + 1
+            ends = start + attention_mask[:, start:].sum(1)
```

Comment:
This will be correct (with the correction to `attention_mask` above), however not for tokenizers which have `<eos> == <pad>`. By construction `attention_mask != <pad>`, so this additional +1 index was getting the last `<eos>` token, which would be treated as padding otherwise.